firebase-perf/src/main/java/com/google/firebase/perf/metrics/AppStartTrace.java

```diff
@@ -395,31 +412,44 @@ private void logAppStartTrace() {
   }
 
   @Override
-  public void onActivityPaused(Activity activity) {
-    if (isExperimentTraceDone()) {
+  public void onActivityPaused(Activity activity) {}
+
+  @Override
+  public void onActivityStopped(Activity activity) {}
+
+  /** App is entering foreground. Keep annotation is required so R8 does not remove this method. */
+  @Keep
+  @OnLifecycleEvent(Lifecycle.Event.ON_START)
+  public void onAppEnteredForeground() {
+    if (isStartedFromBackground || isTooLateToInitUI || firstForegroundTime != null) {
       return;
     }
-    Timer onPauseTime = clock.getTime();
-    TraceMetric.Builder subtrace =
+    // firstForeground is equivalent to the first Activity onStart. This marks the beginning of
+    // observable backgrounding. Prior to this point, backgrounding cannot be observed.
+    firstForegroundTime = clock.getTime();
+    this.experimentTtid.addSubtraces(
```

Comment:
What happens if there are multiple foregrounding/backgrounding during app starts. Would we have a multiple subtraces with the same name as "_experiment_firstforegrounding"/"backgrounding"?

Reply:
If we want to capture all of foregrounds/backgrounds, I would remove the term "first" from the naming. Else, would recommend capturing it just once.

Reply:
This should only capture the first one, anything after would not be added to the trace.